pyiceberg/io/__init__.py

```diff
@@ -316,7 +316,7 @@ def _import_file_io(io_impl: str, properties: Properties) -> Optional[FileIO]:
         class_ = getattr(module, class_name)
         return class_(properties)
     except ModuleNotFoundError:
-        logger.warning("Could not initialize FileIO: %s", io_impl)
```

Comment:
nit: what about something like this

https://github.com/apache/iceberg-python/blob/1e011014cc4d66e12f8d77ce75f9609de9513913/pyiceberg/catalog/__init__.py#L287-L288